### PR TITLE
Fix for public repos, #278

### PR DESCRIPTION
**Pull request by carlocardella**
**This PR fixes Issue#**
Partial fix (public repos) for #278

**Description of the PR**:
This is a partial fix to show public repos, more work needs to be done (likely around the authorization token) to work with private repos

**Additional context**:
Add any other context or screenshots about the PR here.

### CODE REVIEW

**Comment by carlocardella** [comment]
Just noticed that public repos can be listed, but files cannot be opened, needs further investigation: 

`"The editor could not be open due to an unexpected error. Retry"`.

On Retry, the error becomes a generic `Invalid character`

**Comment by carlocardella** [comment]
> Just noticed that public repos can be listed, but files cannot be opened, needs further investigation:
> 
> `"The editor could not be open due to an unexpected error. Retry"`.
> 
> On Retry, the error becomes a generic `Invalid character`

Fixed and pushed; now the problem with the private repos remains

File: src/repos/tree/index.ts [hunk 1]
```diff
@@ -61,7 +61,7 @@ class RepositoryTreeProvider implements TreeDataProvider<TreeItem> {
   getChildren(element?: TreeItem): ProviderResult<TreeItem[]> {
     if (!element && authStore.isSignedIn && store.repos.length > 0) {
       return store.repos
-        .sort((a, b) => a.name.localeCompare(b.name))
+        .slice().sort((a, b) => a.name.localeCompare(b.name))
```
Comment (lostintangent) [inline src/repos/tree/index.ts:64]: This change makes sense, though I'm curious if there was an actual issue that this fixed?

File: src/repos/store/actions.ts [hunk 2]
```diff
@@ -161,7 +159,17 @@ export async function getRepoFile(
   const api = await getApi(GitHub);
 
   const response = await api.get(`/repos/${repo}/git/blobs/${sha}`);
-  return base64ToUintArray(response.body.content);
+
+  return new Uint8Array(
+    Buffer.from(response.body.content, "base64")
```
Comment (lostintangent) [inline src/repos/store/actions.ts:164]: The `Buffer` type isn't available in VS Code for web (e.g. github.dev), so we should probably avoid having to rely on in for any new code.

Is the `base64-to-uint8array` module suddenly not working?

File: src/repos/tree/index.ts (hunk 1 above)
Comment (carlocardella) [reply src/repos/tree/index.ts:64]: It was throwing an exception on this line, it was clear in the debug console: the error message had a suggestion about how to fix it

File: src/repos/store/actions.ts (hunk 2 above)
Comment (carlocardella) [reply src/repos/store/actions.ts:164]: Ah, I have not tested it on VSCode for web; the call behind base64ToUintArray was throwing an "invalid character" error even with a very simple markdown file; unfortunately I was not able to figure out why so I tried to copy that logic (minus the validation error) into this file. It works well on desktop (open files, updates etc.). 

Do you know an alternative to use for the web? I can investigate if needed

https://github.com/mafintosh/base64-to-uint8array/blob/8fe1a40498ac27297fccdb6cd0e8771c47a30c74/index.js#L3

Comment (carlocardella) [reply src/repos/store/actions.ts:164]: I tested it locally, it is possible to replacew `Buffer` with this:

```typescript
  return Uint8Array.from(
    response.body.content.split("").map((letter: string) => letter.charCodeAt(0))
  );
```

Is this supported in VSCode for Web? Can I push this change?

Comment (carlocardella) [reply src/repos/store/actions.ts:164]: I took another look at this, `base64ToUintArray` looks like this:

```typescript
function browser (s) {
  return new Uint8Array(atob(s).split('').map(charCodeAt))
}

function node (s) {
  var b = require('buf' + 'fer')
  return new b.Buffer(s, 'base64')
}
```

`browser` calls `atob` which returns the `invalid character` exception; anyway `atob` internally uses `Buffer`:

```typescript
function atob(input) {
  // The implementation here has not been performance optimized in any way and
  // should not be.
  // Refs: https://github.com/nodejs/node/pull/38433#issuecomment-828426932
  input = `${input}`;
  for (let n = 0; n < input.length; n++) {
    if (!kBase64Digits.includes(input[n]))
      throw lazyDOMException('Invalid character', 'InvalidCharacterError');
  }
  return Buffer.from(input, 'base64').toString('latin1');
}
```

So it looks like `Buffer` is used anyway? Can't we use it directly to bypass the `invalid character` error?

File: src/repos/tree/index.ts (hunk 1 above)
Comment (carlocardella) [reply src/repos/tree/index.ts:64]: You're right, I did further testing last night and the real root cause was the exception from `base64-to-uint8array`. Do you want me to revert this change?

**Comment by carlocardella** [comment]
Abandoning this PR in favor of #284

**Pull request closed without merging**